chapter_1/minisat_installation.md

```diff
@@ -1,6 +1,83 @@
 # Minisat Installation
-+ libvirt API
-+ virt-install
-+ virsh commands
-+ docker engine
-+ docker instance
+
+#### System Requirements
++ 64-bit Architecture
++ A minimum 250GB storage and 4GB memory
++ Developed and Tested on Fedora 27
+
+#### Pre-requisites
++ All system should have Libvirt installed for VM provisioning.
+
+```sh
+dnf install qemu-kvm qemu-img virt-manager libvirt libvirt-python libvirt-client virt-install -y
+```
+
++ All system should have Docker installed for running docker container. To install Docker [follow](https://docs.docker.com/install/linux/docker-ce/fedora/#install-using-the-repository)
+
++ Server should have docker machine installed.
+
+```sh
+curl -L https://github.com/docker/machine/releases/download/v0.13.0/docker-machine-`uname -s`-`uname -m` >/tmp/docker-machine &&
+sudo install /tmp/docker-machine /usr/local/bin/docker-machine
+```
+
++ Server should have HTTP service installed and running.
+
+```sh
+dnf install httpd -y
+```
+
++ Server should have SSH public key or it can be generated using SSH keygen.
+
+```sh
+ssh-keygen
+```
+
+#### Installation
+Minisat uses Python3 and Django web framework so we need to install it. It is recommended to use python3 virtual environment. To create python3 virtual environment
+```sh
+python3 -m venv <environment_name>
+```
+
+After that we need to activate the virtual environment by executing
+```sh
+source <environment_name>/bin/activate
+```
+
+Now clone the github Minisat repository
+```sh
+git clone https://github.com/miniSat/minisat.git
+```
+
+Minisat require some python modules like django (version 2.0) and docker. We can install them by executing
```

Comment:
And testing module like Pytest Also

Reply:
pytest is not required for normal user, it is use only for testing.